framework/include/interfaces/HDGInterface.h

```diff
@@ -0,0 +1,32 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "libmesh/libmesh_config.h"
+#include <Eigen/Dense>
+
+#ifdef LIBMESH_USE_COMPLEX_NUMBERS
+typedef Eigen::MatrixXcd EigenMatrix;
+typedef Eigen::VectorXcd EigenVector;
+#else
+typedef Eigen::MatrixXd EigenMatrix;
+typedef Eigen::VectorXd EigenVector;
+#endif
+
+class HDGInterface
```

Comment:
is interface the right name / system for these classes?
what is it interfacing between? The system and the ROs?

Reply:
It's a common interface for both hybridized kernels and boundary conditions

Reply:
Sure but not every common class has to be called an interface

Reply:
If you mean interface to the Assembly then it's fine. But it could be also `HDGAssemblyBase` or `HDGAssemblyInterface` or `HDGAssemblyHelper`. 

`Base` somewhat implies a lot of the other interfaces will be included
anyway this is a side topic, dont view this as a blocking conversation

Reply:
I kind of like XXXAssemblyInterface

Reply:
In this case I'm going to call it `HDGData` since it just supplies data members and no method interfaces. But for the "interfaces" with meat to them I'm going to go with XXXAssemblyInterface

Reply:
Err, I think I like XXXHDGAssemblyHelper. I'll probably switch again, we'll have to see haha